pkg/kclient/pods.go

```diff
@@ -85,6 +85,63 @@ func (c *Client) WaitAndGetPod(watchOptions metav1.ListOptions, desiredPhase cor
 	}
 }
 
+// WaitAndGetPodOnInitContainerStarted blocks and waits until the named initiContainer is started.
+func (c *Client) WaitAndGetPodOnInitContainerStarted(watchOptions metav1.ListOptions, initContainerName, waitMessage string, hideSpinner bool) (*corev1.Pod, error) {
```

Comment:
I think this function could be reduced by using the code from other functions in this package